### PR TITLE
#580 Show company logo in Header

### PR DESCRIPTION
**Pull request by CarlosNZ**
Implements #580 

Please use back-end branch `#583-final-tidyups-for-create-join-org`

Note, I've also merged front-end `#583-final-tidy-ups-for-create-join-org'`, so there's currently a few little changes showing in the diff that don't belong in this issue -- the only files I've changed here are:
- UserArea.tsx
- utiltityFunctions.ts

To get logos for existing orgs into the database, please use this snapshot, and these files:
[581_logos.graphql.zip](https://github.com/openmsupply/application-manager-web-app/files/6339747/581_logos.graphql.zip) - snapshot 
[files-#580.zip](https://github.com/openmsupply/application-manager-web-app/files/6322123/files-.580.zip) - files folder

Then you should see logos for 3 of the 4 available orgs in the Header.

Note that one of them (Drug Dealers West) has an external URL, and that shows up fine, too -- I wrote a small utility function to check if the URL is an external one or a relative local one.

### CODE REVIEW

File: src/utils/helpers/utilityFunctions.ts [hunk 1]
```diff
@@ -5,3 +5,14 @@ export const isArraySorted = (array: any[], ascending = true) => {
...
 }
+
+/*
+Checks if a URL string is an external URL or a local/relative one
+If local/relative, it prepends it with a specified host, else it
+returns it unchanged.
+*/
+export const getFullUrl = (baseUrl: string | undefined, host: string) => {
+  const fullUrl = baseUrl?.startsWith('http') ? baseUrl : host + baseUrl
+  if (baseUrl?.startsWith('http')) console.log('url', baseUrl)
```
Comment (nmadruga) [inline src/utils/helpers/utilityFunctions.ts:16]: Should this console be removed?

**Review by nmadruga — COMMENTED** [review]
Testing now...

File: src/utils/helpers/utilityFunctions.ts (hunk 1 above)
Comment (CarlosNZ) [reply src/utils/helpers/utilityFunctions.ts]: Yes! Fixed and simplified 47ff6ff4

**Review by nmadruga — CHANGES_REQUESTED** [review]
As discussed in person: couldn't use the `use_snapshot` for testing due to a error during the load of graphql file, so have copied the files content into the folder (on server): `/files` and added the logoUrl to each of the companies as listed in the `graphql` file. Tried to run again server and front-end, but keep getting this problem:

Screen Shot 2021-04-20 at 11 31 31 AM
Screen Shot 2021-04-20 at 11 31 23 AM

**Comment by nmadruga** [comment]
Some conflicts to resolve here too, please.

**Comment by CarlosNZ** [comment]
> As discussed in person: couldn't use the `use_snapshot` for testing due to a error during the load of graphql file, so have copied the files content into the folder (on server): `/files` and added the logoUrl to each of the companies as listed in the `graphql` file. Tried to run again server and front-end, but keep getting this problem:
> 
> Screen Shot 2021-04-20 at 11 31 31 AM
> 
> Screen Shot 2021-04-20 at 11 31 23 AM

I've made a new snapshot and updated the link in the original post. Hopefully works now. 🤞

**Comment by CarlosNZ** [comment]
> Some conflicts to resolve here too, please. I'll have another look on this PR and other ones on the afternoon so hopefully we can merge all of them today \o/

Done f7d545a1

**Comment by nmadruga** [comment]
> > As discussed in person: couldn't use the `use_snapshot` for testing due to a error during the load of graphql file, so have copied the files content into the folder (on server): `/files` and added the logoUrl to each of the companies as listed in the `graphql` file. Tried to run again server and front-end, but keep getting this problem:
> > Screen Shot 2021-04-20 at 11 31 31 AM
> > Screen Shot 2021-04-20 at 11 31 23 AM
> 
> I've made a new snapshot and updated the link in the original post. Hopefully works now. 🤞

Not sure what is happening : ( I tried to reallocate the files, ran the server the `use_snapshot` twice, tried to restart the front-end, git fetch, but still not a sucessfull result:
Screen Shot 2021-04-20 at 4 48 22 PM

**Comment by nmadruga** [comment]
Cool, tested using this steps. And checked the logos are correctly displayed in the User area. ✅ 

>  think you’ll have to do what I did and manually add them:
> - Start a FeatureShowcase application, skip to the end and use it to upload some files (don’t worry about submitting).
> - Look at the “file” table in the database, and copy the unique_id value for each
> - replace company logoUrl fields with /file?uid=<unique_id>

> I just used two files — I added one to two different orgs, and left one org with no logo (so you can see what it does in that case)

**Review by nmadruga — APPROVED** [review]

**Pull request merged**